### PR TITLE
feat(pods): Add pods GitHub enterprise support

### LINKED ISSUE #6568: CocoaPods GitHub enterprise hosted dependency support
**What Renovate type are you using?**
We are currently using renovate-on-prem for our web app

**Describe the bug**
We are looking to use the same in iOS, automate dependency updates for Cocoapods. We noticed an issue with GitHub enterprise hosted dependency (private Cocoapods repo) as they are not updated. Does renovate support CocoaPods GitHub enterprise hosted dependency? Any suggestion or fix for this issue?

**Did you see anything helpful in debug logs?**
`"msg":"Failed to look up dependency ___________"`

**To Reproduce**
1.  Create a private CocoaPods repo on github enterprise (github enterprise and requires auth for read access perhaps)
2. Create a sample iOS project with CocoaPods that has pod dependency that's hosted in private CocoaPods repo on github enterprise
3. Run renovate and notice the error to resolve dependency

**Additional context**

### PR DESCRIPTION
**Pull request by asoneji**
## Changes:

Add pods GitHub enterprise support

## Context:

fixes #6568

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/metadata.ts [hunk 1]
```diff
@@ -110,8 +110,14 @@ const manualSourceUrls = {
...
+  let massagedUrl = url;
+
+  if (url.startsWith('git@')) {
+    massagedUrl = url.replace(':', '/').replace('git@', 'https://');
+  }
+
+  return massagedUrl
     .replace('http:', 'https:')
     .replace('http+git:', 'https:')
     .replace('https+git:', 'https:')
```
Comment (asoneji) [inline lib/datasource/metadata.ts:123]: Thanks Kerry Kartchner for the suggestion.

```suggestion
    .replace('https+git:', 'https:')
    .replace('ssh://git@','https://')
```

File: lib/datasource/pod/index.ts [hunk 2]
```diff
@@ -32,13 +33,15 @@ function shardParts(lookupName: string): string[] {
 
 function releasesGithubUrl(
   lookupName: string,
-  opts: { account: string; repo: string; useShard: boolean }
+  opts: { hostURL: string; account: string; repo: string; useShard: boolean }
 ): string {
-  const { useShard, account, repo } = opts;
-  const prefix = 'https://api.github.com/repos';
+  const { hostURL, account, repo, useShard } = opts;
+  const prefix = hostURL
```
Comment (asoneji) [inline lib/datasource/pod/index.ts:39]: Thanks, Kerry Kartchner for the suggestion: 

```suggestion
  const prefix = hostURL && !hostURL.includes('api.github.com')
```

File: lib/datasource/pod/index.ts [hunk 3]
```diff
@@ -98,14 +101,17 @@ const githubRegex = regEx(
   /^https:\/\/github\.com\/(?<account>[^/]+)\/(?<repo>[^/]+?)(\.git|\/.*)?$/
 );
 
+const githubEnterpriseRegex = regEx(
+  /(?<hostURL>(^https:\/\/github\..*\.com))\/(?<account>[^/]+)\/(?<repo>[^/]+?)(\.git|\/.*)?$/
```
Comment (asoneji) [inline lib/datasource/pod/index.ts:105]: Thanks, Kerry Kartchner for the suggestion.
```suggestion
  /(?<hostURL>(^https:\/\/github\..+\.[a-zA-z0-9-]+))\/(?<account>[^/]+)\/(?<repo>[^/]+?)(\.git|\/.*)?$/
```

File: lib/datasource/pod/index.ts [hunk 4]
```diff
@@ -98,14 +101,17 @@ const githubRegex = regEx(
...
+);
+
 async function getReleasesFromGithub(
   lookupName: string,
-  registryUrl: string,
+  match: RegExpExecArray,
```
Comment (asoneji) [inline lib/datasource/pod/index.ts:110]: Thanks, Kerry Kartchner for the suggestion.

```suggestion
  opts: { hostURL: string; account: string; repo: string; useShard: boolean }
```

**Review by viceice — CHANGES_REQUESTED** [review]
You need to fix the git author and/or committer.

**Comment by asoneji** [comment]
My git config for github.com was not set up correctly. I close this PR and resolved the issue in https://github.com/renovatebot/renovate/pull/12818.

**Comment by viceice** [comment]
no, you can simply force push ypour changes

**Comment by asoneji** [comment]
I had tried that but was not able to reset one of the commits with the correct author. This is resolved in the new PR thanks.

**Pull request closed without merging**